common/src/main/java/dev/architectury/hooks/item/tool/HoeItemHooks.java

```diff
@@ -0,0 +1,23 @@
+package dev.architectury.hooks.item.tool;
+
+import com.google.common.collect.ImmutableMap;
+import com.mojang.datafixers.util.Pair;
+import net.minecraft.world.item.HoeItem;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.block.Block;
+
+import java.util.HashMap;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public class HoeItemHooks {
+    private HoeItemHooks() {
+    }
+
+    public static void addTillingRecipe(Block src, Predicate<UseOnContext> predicate, Consumer<UseOnContext> action) {
```

Comment:
Javadoc is especially needed here - what does the predicate do? What does the action do?

Reply:
TL;DR is: Game invokes `predicate.test(ctx)`, if that returns true the game first invokes `action.accept(ctx)` and then damages the hoe by 1.